config/src/main/java/org/springframework/security/config/annotation/web/AbstractRequestMatcherRegistry.java

```diff
@@ -206,6 +210,35 @@ public C regexMatchers(String... regexPatterns) {
 		return chainRequestMatchers(RequestMatchers.regexMatchers(regexPatterns));
 	}
 
+	/**
+	 * Maps a {@link List} of
+	 * {@link org.springframework.security.web.util.matcher.DispatcherTypeRequestMatcher}
+	 * instances.
+	 * @param method the {@link HttpMethod} to use or {@code null} for any
+	 * {@link HttpMethod}.
+	 * @param dispatcherTypes the dispatcher types to match against
+	 * @return the object that is chained after creating the {@link RequestMatcher}
+	 */
+	public C dispatcherTypeMatchers(HttpMethod method, DispatcherType... dispatcherTypes) {
+		Assert.state(!this.anyRequestConfigured, "Can't configure dispatcherTypeMatchers after anyRequest");
+		return chainRequestMatchers(Arrays.stream(dispatcherTypes)
```

Comment:
Please use a for-loop instead of a Stream. 
The GC overhead of creating additional objects (including intermediate objects) can cause significant decrease in performance.
See #7154 for more details.